### PR TITLE
Make forum wiki articles available in Map

### PR DESCRIPTION
**Pull request by Bouncey**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX

### TODO
- [x] Create an article component to display the article
- [x] create routing logic that maps to `/wiki/${topic.dashedName}`
- [x] refactor to interface better with `makePanelOpen/Hidden` selectors
- [x] make searching possible

### CODE REVIEW

**Comment by BerkeleyTrue** [comment]
From my conversations with @QuincyLarson the wiki content should be shown along with the challenges.

Also, we shouldn't be requesting wiki content using server to server but instead be caching it somehow and regularly updating our cache.

**Comment by Bouncey** [comment]
I thought the idea was to have the articles in the Map, and that they are searchable from the same input as the challenges

I have an instance of mongo on mlab holding the articles and updating itself every day. I had to build my server as something that can pull data from the discourse api, then present the data as required for each request.

With the discourse api, it is not as simple as requesting all topics/articles in a category/sub-category. First you have to request a list of topics, which is paginated with an offset of 30 (there are ~780 topics in the wiki category, including sub-categories), then you have to make a single request for each article. 

The code for the server can be found [here](https://github.com/Bouncey/bounceyBot) (it has some Rx in it :wink: )

**Comment by QuincyLarson** [comment]
@Bouncey excellent - ideally, we'll want to paginate through these results and then cache them somewhere at a 24-hour interval, then do the same with YouTube and potentially Medium (I'm talking with them trying to figure out a good way to do this).

**Comment by QuincyLarson** [comment]
I tried accessing it and I'm getting this issue:
 is there a quick fix for this?

**Comment by Bouncey** [comment]
This PR shouldn't have an effect on that error. #12016 is to reduce those
errors, but I will need to refactor for issues with the map.

As for this PR, you should be getting some sort of response from my server,
try the network tab and look for a response from
https://glitter-organisation.gomix.me/api/v1/wiki

I've just checked it and I am getting an error response. I will look into
it when I get home.

**Comment by BerkeleyTrue** [comment]
@QuincyLarson You can ignore those errors

**Comment by QuincyLarson** [comment]
@Bouncey have you made any progress on this? This still isn't working on my end. Eager to QA this and get this live on beta :)

**Comment by Bouncey** [comment]
I've been slaughtered at work, I had an issue with Rx but I think I have worked my way though it, should be working after my next commit today or tomorrow

**Comment by Bouncey** [comment]
Still to implement:

Interfacing with `makePanelOpenHidden` selectors,
Searching

After adding the wiki to the mapUi node tree, the performance drops, on my machine anyway. I might not be updating the tree properly, which could cause my issues.

@BerkeleyTrue @QuincyLarson  feedback is much appreciated please.

File: server/utils/index.js [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
+function unDasherize(name) {
   return ('' + name)
     // replace dash with space
     .replace(/\-/g, ' ')
     // strip nonalphanumarics chars except whitespace
     .replace(/[^a-zA-Z\d\s]/g, '')
     .trim();
 }
+
+module.exports = {
```
Comment (Bouncey) [inline server/utils/index.js]: I hit the same issue I had in a previous PR of not being able to use this in the app, using `module.exports` resolved the issue.

Comment (BerkeleyTrue) [inline server/utils/index.js]: Can you explain the issue?

Comment (Bouncey) [inline server/utils/index.js]: `index.js:1 Uncaught SyntaxError: Unexpected token export`

The console points to `export` on line 1 of this file. 

The error doesn't make any sense to me. 

If I use `module.exports`, there is no issue

Comment (BerkeleyTrue) [inline server/utils/index.js]: @Bouncey That error usually means that the node does not know how to interpret a line of code. Doesn't sound like an issue in code but an issue with your set up. Which version of node are you running?

Comment (Bouncey) [inline server/utils/index.js]: `6.9.1`

Comment (Bouncey) [inline server/utils/index.js]: I bumped my node version up to 7.5 and still see the same error

File: webpack.config.js [hunk 2]
```diff
@@ -31,7 +31,8 @@ module.exports = {
         test: /\.jsx?$/,
         include: [
           path.join(__dirname, 'client/'),
-          path.join(__dirname, 'common/')
+          path.join(__dirname, 'common/'),
+          path.join(__dirname, 'server/')
```
Comment (Bouncey) [inline webpack.config.js]: @BerkeleyTrue this has solved my syntax errors for import/export from server files

Comment (BerkeleyTrue) [inline webpack.config.js]: I see the issue now. Yes we should be including server files in webpack's loaders. Instead we can relocate those utils into the common directory.

**Pull request closed without merging**